packages/blockchain/src/db/manager.ts

```diff
@@ -166,8 +160,12 @@ export class DBManager {
         throw error
       }
     }
-    const blockData = [header, ...body] as BlockBuffer
-    const opts = { common: this._common, hardforkByBlockNumber: true }
+    const blockData = [header.raw(), ...body] as BlockBuffer
+    let parentTd
+    if (!number.eqn(0)) {
+      parentTd = await this.getTotalDifficulty(header.parentHash, number.subn(1))
+    }
+    const opts = { common: this._common, hardforkByTD: parentTd }
```

Comment:
Shouldn't `hardforkByTD: parentTd` instead be `hardforkByTD: parentTd.add(header.difficulty)`?

My reasoning is that the current totalDifficulty of the block (and the one that's used for the fork choice) is the parent total difficulty + the current block difficulty.

Reply:
Then we wouldn't need to get the parent TD in the first place and could directly take the one from the block. 😀

But I was confused by this as well and actually first had this differently. Still a bit unsure but I guess it doesn't make sense to interpret in a way that the - theoretical - POW difficulty of a block should still be counted in to then determine the switch of the very same block to POS and then rather set this difficulty to 0.

So I guess this has to be the parent TD by definition. Let me know if I'm missing something.

Reply:
Oh, I'm probably missing something basic then... Is this always going to be called in the context of PoS? Your answer seems to imply that but I can't see why that would be the case in the code.

Reply:
Hmm, what do you mean with "context of PoS"? This is just trying to determine the correct HF for a block or header. The `hardforkByTD` option (respectively the total difficulty part of it) is generally - practically - only relevant for now for the Merge HF switch, is this what you mean with PoS context?

And the total difficulty provided for this option should just be the correct total difficulty which is used (per spec) for the decision if the block we be seen as a PoS block. I am arguing that this must be logically the parent TD, but I might also be missing something here.